dbee/adapters/databricks_driver.go

```diff
@@ -0,0 +1,130 @@
+package adapters
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/url"
+
+	"github.com/kndndrj/nvim-dbee/dbee/core"
+	"github.com/kndndrj/nvim-dbee/dbee/core/builders"
+)
+
+var (
+	_ core.Driver           = (*databricksDriver)(nil)
+	_ core.DatabaseSwitcher = (*databricksDriver)(nil)
+)
+
+// databricksDriver is a driver for Databricks.
+type databricksDriver struct {
+	// c is the client used to execute queries.
+	c *builders.Client
+	// connectionURL is the parsed connection URL.
+	// a DSN structure in the format of:
+	// token:[my_token]@[hostname]:[port]/[endpoint http path]?param=value
+	connectionURL *url.URL
+	// currentCatalog is the current catalog.
+	// Retrieved from the connectionURL parameters.
+	currentCatalog string
+}
+
+// Query executes the given query and returns the result stream.
+func (d *databricksDriver) Query(ctx context.Context, query string) (core.ResultStream, error) {
+	return d.c.QueryUntilNotEmpty(ctx, query)
+}
+
+// Columns returns the columns for the given table.
+func (c *databricksDriver) Columns(opts *core.TableOptions) ([]*core.Column, error) {
+	return c.c.ColumnsFromQuery(`
+		SELECT column_name, data_type
+		FROM information_schema.columns
+		WHERE
+			table_schema='%s' AND
+			table_name='%s';`,
+		opts.Schema, opts.Table)
+}
+
+// Structure returns the structure of the current catalog/database.
+func (d *databricksDriver) Structure() ([]*core.Structure, error) {
+	catalogQuery := fmt.Sprintf(`
+SELECT table_schema, table_name, table_type
+FROM system.information_schema.tables
+WHERE table_catalog = '%s';
+`, d.currentCatalog)
+
+	rows, err := d.Query(context.Background(), catalogQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	return core.GetGenericStructure(rows, getDatabricksStructureType)
+}
+
+// getDatabricksStructureType returns the core.StructureType based on the
+// given type string for databricks adapter.
+func getDatabricksStructureType(typ string) core.StructureType {
+	switch typ {
+	case "TABLE", "BASE TABLE", "SYSTEM TABLE":
+		return core.StructureTypeTable
+	case "VIEW", "SYSTEM VIEW":
+		return core.StructureTypeView
+	case "MATERIALIZED_VIEW":
+		return core.StructureTypeMaterializedView
+	case "STREAMING_TABLE":
+		return core.StructureTypeStreamingTable
+	case "MANAGED":
+		return core.StructureTypeManaged
+	default:
+		return core.StructureTypeNone
+	}
+}
```

Comment:
@MattiasMTS, these new types are not supported in the UI. The tree is, e.g., not displaying objects `"managed"`. A simple solution would be to consider there as `"table"`, which would make the core code more modular since these are in `Unity Catalog` terms considered to be just that; `tables`.
```suggestion
func getDatabricksStructureType(typ string) core.StructureType {
	switch typ {
	case "TABLE", "BASE TABLE", "SYSTEM TABLE", "MANAGED", "STREAMING_TABLE":
		return core.StructureTypeTable
	case "VIEW", "SYSTEM VIEW", "MATERIALIZED_VIEW":
		return core.StructureTypeView
	default:
		return core.StructureTypeNone
	}
}

```
This could also be handled on the core side as well, but think for standardisation reasons it makes sense to use the `StructureType`s already in place.

Reply:
sure I can fix this.

Reply:
Thanks, also noticed that there are more type, e.g, `MANAGED_SHALLOW_CLONE`, that should probably be included in the `table` category. I haven't found a complete list of table types, but an idé would be to have `*MANAGED*` be a wildcard for tables, and `*VIEW*` for view's. But The above should be sufficient for now I suppose.